### ISSUE TITLE
Resolve issue with hydra building PR

### ISSUE THREAD

**Issue by samrose** [Holo-Host/holo-nixpkgs#311]
Hydra does not build PR's

We need to diagnose and fix the issue so that hydra creates jobsets and builds PR's

**Comment by samrose** [comment]
The problem is resolved with re-opening existing PR, but not with opening new PR

I will need to set up a development clone of holo-nixpkgs that I can continue this testing on. So I will do that next.

**Comment by yegortimoshenko** [comment]
> I will need to set up a development clone of holo-nixpkgs that I can continue this testing on. So I will do that next.

Please don't. For purposes of what you're doing, a hello world repo would be enough.

**Comment by yegortimoshenko** [comment]
Please also revert things back into working state. `hydra.holo.host` was pointing to 68bc1d2b1bced406d00648f11e9bb75b2e99050d, which is a working rev. Now, it's pointing to 74f8b70af1bc62366298e85a85c5f5d38688a626.

**Comment by samrose** [comment]
> Please also revert things back into working state. `hydra.holo.host` was pointing to [68bc1d2](https://github.com/Holo-Host/holo-nixpkgs/commit/68bc1d2b1bced406d00648f11e9bb75b2e99050d), which is a working rev. Now, it's pointing to [74f8b70](https://github.com/Holo-Host/holo-nixpkgs/commit/74f8b70af1bc62366298e85a85c5f5d38688a626), which is broken.

I logged in and observed that you hav edone this.

**Comment by samrose** [comment]
> > I will need to set up a development clone of holo-nixpkgs that I can continue this testing on. So I will do that next.
> 
> Please don't. It should be a minimal reproducible example. For purposes of what you're doing, a hello world repo with the most stripped down `release.nix` would be enough.

If you want to assert how people do work, you should do that earlier in the arc of their work on something.

**Comment by samrose** [comment]
> > I will need to set up a development clone of holo-nixpkgs that I can continue this testing on. So I will do that next.
> 
> Please don't. It should be a minimal reproducible example. For purposes of what you're doing, a hello world repo with the most stripped down `release.nix` would be enough.

From what I can see, the other paradox that this puts me into is that since hydra is controlled by holo-nixpkgs, I need to have a `profiles/logical/holo/hydra/master/specs/<myproject>.json` present in holo-nixpkgs for my "example" project, in order to replicate the problem. But that won't be possible when I am testing older commits of the same repository that do not possess this hydra spec file.

**Comment by yegortimoshenko** [comment]
> From what I can see, the paradox that this puts me into is that since hydra is controlled by holo-nixpkgs, I need to have a `profiles/logical/holo/hydra/master/specs/<myproject>.json` present in holo-nixpkgs for my "example" project, in order to replicate the problem. But that won't be possible when I am testing older commits of the same repository that do not possess this hydra spec file. 

I'm not sure I understand the issue.

Essentially:

1. Create `hydra-example` repo in Holo organization.
2. Add `nixpkgs.nix` to the repo:
    ```nix
    import (fetchTarball {
      url = "https://github.com/Holo-Host/holo-nixpkgs/archive/fc160c13bd2f88409d09866ea59945b9579dd23a.tar.gz";
      sha256 = "09wsb9n62rxgr4pqz2z1098xcj8fqwxnq0iv0lhprj15vbkxpgih";
    })
    ```
2. Add `release.nix` to the repo:
    ```nix
    { pkgs ? import ./nixpkgs.nix {} }: {}
    ```
2. Add `jobsets.nix` to the repo:
    ```nix
    { pkgs ? import ./nixpkgs.nix {} }:

    with pkgs;

    mkJobsets {
      owner = "Holo-Host";
      repo = "hydra-example";
      branches = [ "develop" "master" ];
      pullRequests = <hydra-example-pull-requests>;
    }
    ```
2. Create `profiles/logical/holo/hydra/master/specs/hydra-example.json` by copying `holo-nixpkgs.json` to that location and running `s/holo-nixpkgs/hydra-example/g` in the new file.
3. Create new declarative project on https://hydra.holo.host, named `hydra-example`, specifying `profiles/logical/holo/hydra/master/specs/hydra-example.json` as "Declarative spec file", and "Git checkout" + "https://github.com/Holo-Host/holo-nixpkgs.git develop" as "Declarative input type".

---

1. Clone Holo Nixpkgs locally to `hydra.holo.host`.
2. Enter `nix-shell` inside the checkout.
3. Read https://git-scm.com/docs/git-bisect.
3. Run `git bisect start 74f8b70af1bc62366298e85a85c5f5d38688a626 68bc1d2b1bced406d00648f11e9bb75b2e99050d`.
3. On each rev, run `sudo nixos-rebuild switch` and open a PR against `hydra-example` repo. If you see your PR in https://hydra.holo.host/project/holo-nixpkgs, run `git bisect good`. If not, `git bisect old`.

**Comment by yegortimoshenko** [comment]
> If you want to assert how people do work, you should do that earlier in the arc of their work on something.

This is a common sense request on my part: https://stackoverflow.com/help/minimal-reproducible-example

This also presumes everything has to go through me, which isn't (and has been) unsustainable for me.

**Comment by samrose** [comment]
>This also presumes everything has to go through me, which isn't (and has been) unsustainable for me.

No. It presumes that if you want to assert how people should do work, you should do it earlier in arc of their work.

**Comment by samrose** [comment]
> 6\. Create new declarative project on https://hydra.holo.host, named `hydra-example`, specifying `profiles/logical/holo/hydra/master/specs/hydra-example.json` as "Declarative spec file", and "Git checkout" + "https://github.com/Holo-Host/holo-nixpkgs.git develop" as "Declarative input type".

Sorry, this was the part where I did not realize where exactly hydra would read this file from. This makes sense. Thank you.

**Comment by samrose** [comment]
> This is a common sense request on my part: https://stackoverflow.com/help/minimal-reproducible-example

You are the first person I've worked with in 25+ years of doing this type of work that has asserted something like this. This is just a fact.


'Minimal, Reproducible Example" is definitely a good work practice. It is even sensible. But I disagree that it is "common".

**Comment by samrose** [comment]
> Clone Holo Nixpkgs locally to hydra.holo.host.
> Enter nix-shell inside the checkout.

@yegortimoshenko It's not clear to me how cloning and entering nix-shell will result in the hydra.holo.host server now serving my cloned repo vs the repo it is set up to serve via nix-channel

Are you actually stating here that just simply cloning the repo, and then running nix-shell will result in the nixos server to be serving the cloned repo vs what it had built and served from the `holo-nixpkgs` channel it is subscribed to? I'll give it a try but thought I'd ask for clarity

**Comment by yegortimoshenko** [comment]
Just like `hpos-shell`, `hpos-switch` inside Holo Nixpkgs `nix-shell` will use your Holo Nixpkgs clone, instead of system-wide Holo Nixpkgs checkout.

See: https://github.com/Holo-Host/holo-nixpkgs/blob/develop/shell.nix#L27

**Comment by filalex77** [comment]
Fixed in #448.
